tests/sibernetic/generate_wcon.py

```diff
@@ -0,0 +1,187 @@
+                          (middle_point, middle_points[-1][0],
+                           middle_points[-1][1], xs[middle_point],
+                           ys[middle_point], middle_point_speed_x[-1],
+                           middle_point_speed_y[-1]))
+
+                    print("  Speed of av point " +
+                          "(%s,%s) -> : (%s,%s): x %sum/s, y %sum/s" %
+                          (ave_points[-1][0], ave_points[-1][1], avx, avy,
+                           ave_point_speed_x[-1], ave_point_speed_y[-1]))
+
+                middle_points.append(
+                    (xs[middle_point] - offset, ys[middle_point]))
+                ave_points.append((avx, avy))
+                time_points.append(t_s)
+
+                print("  Plot frame %i at %s s; l %i: [(%s,%s),...#%i]\n" % (
+                    num_plotted_frames, t_s, line_num, xs[0], ys[0], len(xs)))
+                ax.plot(xs, ys, '-')
+                num_plotted_frames += 1
+                if num_plotted_frames % 5 == 1:
+                    time = '%ss' % t_s if not t_s == int(t_s) \
+                        else '%ss' % int(t_s)
+                    ax.text(50 + ((num_plotted_frames - 1) * spacing),
+                            10, time, fontsize=12)
+
+    info = "Loaded: %s points from %s, saving %i frames" % (
+        line_num, pos_file_name, num_frames)
+    print(info)
+
+    wcon = open(wcon_file_name, 'w')
+
+    wcon.write('''{
+    "units":{ "t":"s - check with Andrey!!",
+              "x":"um",
+              "y":"um"},
+    "comment":"Saved from Sibernetic data.",
+    "note":"%s",
+    "data":[\n''' % info)
+
+    for t in ts:
+        wcon.write('''
+           {"id":1, "t":%s,
+            "x":%s,
+            "y":%s
+           }''' % (t, x[t], y[t]))
+        if t == ts[-1]:
+            wcon.write('\n        ]')
+        else:
+            wcon.write(',\n')
+
+    wcon.write('\n}\n')
+    wcon.close()
+    postions_file.close()
+
+    if plot:
+
+        fig = plt.figure()
+        plt.plot(time_points[1:], middle_point_speed_x, 'cyan',
+                 label='Speed in x dir of point %i/%i' % (middle_point,
+                                                          points))
+        plt.plot(time_points[1:], middle_point_speed_y, 'red',
+                 label='Speed in y dir of point %i/%i' % (middle_point,
+                                                          points))
+        plt.plot(time_points[1:], ave_point_speed_x, 'blue',
+                 label='Speed in x of average of %i points' % points)
+        plt.plot(time_points[1:], ave_point_speed_y, 'green',
+                 label='Speed in y of average of %i points' % points)
+        plt.legend()
+        plt.show()
+
+
+if __name__ == '__main__':
+
+    if len(sys.argv) == 2:
+        pos_file_name = sys.argv[1]
+    else:
+        pos_file_name = 'crawling_at_agar.txt'
+
+    generate_wcon(pos_file_name, "sibernetic_test_full.wcon", rate_to_plot=5)
```

Comment:
And this name could be derived from the input name, and then it would be a full-featured converter rather than a test example!